overlord/snapstate/snapstate.go

```diff
@@ -786,6 +797,20 @@ func doUpdate(st *state.State, names []string, updates []*snap.Info, params func
 		}
 		ts.JoinLane(st.NewLane())
 
+		if update.Type == snap.TypeOS || update.Type == snap.TypeBase {
```

Comment:
Can you please add a comment that this loop iterates over updates ordered by the priority and this makes this newly added code safe, that is, that we always have prerequisites by the time we need to read them.

Reply:
that's was  the intention of the comments inside the two if sides, but it seems they are not clear enough